docs/reference.md

```diff
@@ -303,6 +311,71 @@ Specify the config file to write to. Without this option, your configuration
 will be written to `~/.gemstash/config.yml`. If you write to a custom location,
 you will need to pass the `--config-file` option to all Gemstash commands.
 
+## Preload
+
+Preload all the gems in your gemstash server from the default upstream.
+This can be useful if you plan to go somewhere where the internet connection
+will not be reliable or available.
+It is worth noting that this will require a large amount of disk space and
+computing time.
+
+### Usage
+
+```
+gemstash preload
+gemstash preload --latest
+gemstash preload --limit=1000
+gemstash preload --limit=1000 --skip=1000
+gemstash preload --threads=10 --server_url=http://my-server-url:9292
+```
+
+### Options
+
+#### --latest
+
+**Usage:** `--latest`
+
+**Description**<br />
+Only fetch the latest specs. This will limit the amount of disk space required,
+but there is no guarantee that the gems you are using will have all their
+dependencies met.
+
+#### --skip
+
+**Usage:** `--skip=<number>`
+
+**Description**<br />
+Number of gems to skip when preloading. This can be useful to load in chunks or
+to resume a gem preloading.
+
+#### --limit
+
+**Usage:** `--limit=<number>`
+
+**Description**<br />
+Limit for the number of gems to preload. In conjunction with [--skip](#--skip)
+this can be used for loading gems in chunks or resuming the process.
```

Comment:
This will download the full specs and go through that list. There is a chance that if you stop the process and restart you may skip less gems if another gem was added, because as you say the gems are sorted alphabetically. But I don't think we should worry too much about it.

One thing I was thinking that may be a cool feature is to provide a flag to *match* gem names and maybe versions, so we could fetch all the *rails* or *active.+* in one go.

But I wanted to keep this simple for now and just push the feature up.